framework/src/main/java/org/checkerframework/framework/ajava/StringLiteralConcatenateVisitor.java

```diff
@@ -0,0 +1,38 @@
+package org.checkerframework.framework.ajava;
+
+import com.github.javaparser.ast.expr.BinaryExpr;
+import com.github.javaparser.ast.expr.StringLiteralExpr;
+import com.github.javaparser.ast.visitor.VoidVisitorAdapter;
+
+/**
+ * A visitor that combines added String literals to their concatenation. For example, the expression
+ * {@code "a" + "b"} becomes {@code "ab"}. This occurs even if when reading from left to right that
+ * the two string literals are not added directly. For example, in the expression {@code 1 + "a" +
+ * "b"} we might imagine that {@code 1 + "a"} is evaluated and then {@code "b"} is added.
+ * Regardless, this is transformed into {@code 1 + "ab"}.
+ */
+public class StringLiteralConcatenateVisitor extends VoidVisitorAdapter<Void> {
+    @Override
+    public void visit(BinaryExpr node, Void p) {
+        super.visit(node, p);
+        if (node.getOperator() == BinaryExpr.Operator.PLUS
+                && node.getRight().isStringLiteralExpr()) {
```

Comment:
By experimentation, I verified that javac returns the literal string "ab" for this code:
```java
    return ("a") + ("b");
```

Please make this implementation handle parenthesization, and add a test.

Reply:
I observed that optimization in the generated bytecode.  But @jwaataja does not observe that in the AST.  So nothing needs to be done here (except maybe add a comment explaining why this code does not look within parentheses).